example/src/main/java/com/stripe/example/activity/SetupIntentActivity.java

```diff
@@ -0,0 +1,231 @@
+                );
+        mCompositeDisposable.add(disposable);
+    }
+
+    private void onCreatedSetupIntent(@NonNull ResponseBody responseBody) {
+        try {
+            final JSONObject jsonObject = new JSONObject(responseBody.string());
+            mSetupIntentValue.setText(jsonObject.toString());
+            mClientSecret = jsonObject.getString("secret");
+            mCreatePaymentMethod.setEnabled(mClientSecret != null);
+        } catch (IOException | JSONException exception) {
+            Log.e(TAG, exception.toString());
+        }
+    }
+
+    private void retrieveSetupIntent() {
+        final Observable<SetupIntent> setupIntentObservable = Observable.fromCallable(
+                () -> mStripe.retrieveSetupIntentSynchronous(
+                        SetupIntentParams.createRetrieveSetupIntentParams(mClientSecret)));
+
+        final Disposable disposable = setupIntentObservable
+                .subscribeOn(Schedulers.io())
+                .doOnSubscribe((d) ->
+                        mProgressDialogController.show(R.string.retrieving_setup_intent))
+                .doOnComplete(() -> mProgressDialogController.dismiss())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(
+                        setupIntent -> mSetupIntentValue.setText(setupIntent != null ?
+                                new JSONObject(setupIntent.toMap()).toString() :
+                                getString(R.string.error_while_retrieving_setup_intent)),
+                        throwable -> Log.e(TAG, throwable.toString())
+                );
+        mCompositeDisposable.add(disposable);
+    }
+
+    private void createPaymentMethod(@NonNull final Card card) {
+        final PaymentMethodCreateParams paymentMethodCreateParams =
+                PaymentMethodCreateParams.create(card.toPaymentMethodParamsCard(), null);
+
+        final Observable<PaymentMethod> paymentMethodObservable = Observable.fromCallable(
+                () -> mStripe.createPaymentMethodSynchronous(paymentMethodCreateParams,
+                        PaymentConfiguration.getInstance().getPublishableKey()));
+
+        final Disposable disposable = paymentMethodObservable
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .doOnSubscribe((d) ->
+                        mProgressDialogController.show(R.string.creating_payment_method))
+                .doOnComplete(() ->
+                        mProgressDialogController.dismiss())
+                .subscribe(
+                        paymentMethod -> {
+                            if (paymentMethod != null) {
+                                mSetupIntentValue
+                                        .setText(paymentMethod.id);
+
+                                mPaymentMethod = paymentMethod;
+                                mConfirmSetupIntent.setEnabled(mPaymentMethod != null);
```

Comment:
i think this and on line 185 are both always true, since we've already gated `paymentMethod != null` on line 179? should this be outside of the if statement?